Classes/Telemetry/BITChannel.m

```diff
@@ -230,19 +246,19 @@ - (void)startTimer {
   if ([self timerIsRunning]) {
     [self invalidateTimer];
   }
-  
+
   self.timerSource = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, self.dataItemsOperations);
   dispatch_source_set_timer(self.timerSource, dispatch_walltime(NULL, NSEC_PER_SEC * self.batchInterval), 1ull * NSEC_PER_SEC, 1ull * NSEC_PER_SEC);
   __weak typeof(self) weakSelf = self;
   dispatch_source_set_event_handler(self.timerSource, ^{
     typeof(self) strongSelf = weakSelf;
     if (strongSelf) {
-        if (strongSelf->_dataItemCount > 0) {
-            [strongSelf persistDataItemQueue];
-        } else {
-            strongSelf.channelBlocked = NO;
-        }
-        [strongSelf invalidateTimer];
+      if (strongSelf->_dataItemCount > 0) {
```

Comment:
The `dataItemCount` can be made thread-safe by only reading or modifying it on the queue used by the timer, as long as that queue is serial. (If not, use a barrier block to make the modification thread-safe.) A separate locking mechanism is not required.